pkg/clusteragent/admission/common/label_selectors.go

```diff
@@ -19,11 +19,7 @@ func DefaultLabelSelectors(useNamespaceSelector bool) (namespaceSelector, object
 
 	if pkgconfigsetup.Datadog().GetBool("admission_controller.mutate_unlabelled") ||
 		pkgconfigsetup.Datadog().GetBool("apm_config.instrumentation.enabled") ||
-		pkgconfigsetup.Datadog().GetBool("admission_controller.inject_config.enabled") ||
-		pkgconfigsetup.Datadog().GetBool("admission_controller.inject_tags.enabled") ||
-		len(pkgconfigsetup.Datadog().GetStringSlice("apm_config.instrumentation.enabled_namespaces")) > 0 ||
-		len(pkgconfigsetup.Datadog().GetStringSlice("admission_controller.inject_config.enabled_namespaces")) > 0 ||
-		len(pkgconfigsetup.Datadog().GetStringSlice("admission_controller.inject_tags.enabled_namespaces")) > 0 {
+		len(pkgconfigsetup.Datadog().GetStringSlice("apm_config.instrumentation.enabled_namespaces")) > 0 {
```

Comment:
Now that the 3 webhooks are independent, I think that this function should not check `apm_config.instrumentation.enabled_namespaces` and `apm_config.instrumentation.enabled`.
I think it should only check `admission_controller.mutate_unlabelled` and be called from the config and the tagsfromlabels webhooks.

I think `LabelSelectors` in the APM webhook should stop using this function and define its own using `apm_config.instrumentation.enabled` and `apm_config.instrumentation.enabled_namespaces` and maybe other things that only apply to APM.

Reply:
Spoke on Slack, we're going to follow up on this change in a future PR but before the next agent release. We will need to modify this to support the target based configuration, which will be a good opportunity to do so.